bcipy/helpers/triggers.py

```diff
@@ -638,3 +639,185 @@ def read_triggers(triggers_file: TextIO) -> List[Tuple[str, str, float]]:
     for _, (name, trg_type, stamp) in enumerate(records):
         corrected.append((name, trg_type, float(stamp) + offset))
     return corrected
+
+
+class TriggerType(Enum):
+    """
+    Enum for the primary types of Triggers.
+    """
+
+    NONTARGET = "nontarget"
+    TARGET = "target"
+    FIXATION = "fixation"
+    PROMPT = "first_pres_target"
+
+    @classmethod
+    def list(cls):
+        """Returns all enum values as a list"""
+        return list(map(lambda c: c.value, cls))
+
+    def __str__(self):
+        return f'{self.value}'
+
+
+class Trigger:
```

Comment:
We should make `Trigger` immutable to retain the current behavior. It seems like a data class would work well here:

    from dataclasses import dataclass
    @dataclass(frozen=True)
    class Trigger:
        label: str
        type: TriggerType
        time: float

Reply:
Is it enough to simply add the `@dataclass(frozen=True)` before the class definition as such (after importing dataclass, of course):
```
@dataclass(frozen=True)
class Trigger:
    """
    Object that encompasses data for a single trigger instance.
    """

    def __init__(self,
                 label: str,
                 type: TriggerType,
                 time: str):
        self.label = label
        self.type = type
        self.time = time
```
Or do I have to add the class attributes individually, as you wrote them? I haven't worked with this functionality before, so I'm unfamiliar with how it works.

Reply:
Also, I ran the `test_triggers.py` file after making these changes, and it looks like freezing the class is causing issues with the unit tests. Given my unfamiliarity, I'm going to remove the immutability feature for the time being, if that's alright.

Reply:
I'm okay leaving this to later, but if you're interested the Python documentation (https://docs.python.org/3/library/dataclasses.html) has some good information on data classes. I think the errors you were getting in the tests were caused by having an `init` method. If you define the class like I have in my comment above things should work, with the exception of the `test_print_trigger`, since dataclasses define their own `__repr__`, which is similar but not exactly like the definition you provided.

Reply:
Looks like it worked, thanks! The `__repr__` looks like it still works, based on the documentation it looks like the current `__repr__` just overrides the `__repr__()` the `dataclass` would create.